build/linux/installer/conf/telegraf.conf

```diff
@@ -703,6 +703,47 @@
   insecure_skip_verify = true
   #tagexclude = ["AgentVersion","AKS_RESOURCE_ID","ACS_RESOURCE_NAME", "Region", "ClusterName", "ClusterType", "Computer", "ControllerType"]
 
+##npm
+[[inputs.prometheus]]
+  #name_prefix="container.azm.ms/"
+  ## An array of urls to scrape metrics from.
+  urls = ["$AZMON_INTEGRATION_NPM_METRICS_URL_LIST_NODE"]
+  
+  metric_version = 2
+  url_tag = "scrapeUrl"
+
+  ## An array of Kubernetes services to scrape metrics from.
+  # kubernetes_services = ["http://my-service-dns.my-namespace:9100/metrics"]
+
+  ## Kubernetes config file to create client from.
+  # kube_config = "/path/to/kubernetes.config"
+
+  ## Scrape Kubernetes pods for the following prometheus annotations:
+  ## - prometheus.io/scrape: Enable scraping for this pod
+  ## - prometheus.io/scheme: If the metrics endpoint is secured then you will need to
+  ##     set this to `https` & most likely set the tls config.
+  ## - prometheus.io/path: If the metrics path is not /metrics, define it with this annotation.
+  ## - prometheus.io/port: If port is not 9102 use this annotation
+  # monitor_kubernetes_pods = true
+
+  ## Use bearer token for authorization. ('bearer_token' takes priority)
+  bearer_token = "/var/run/secrets/kubernetes.io/serviceaccount/token"
+  ## OR
+  # bearer_token_string = "abc_123"
+
+  ## Specify timeout duration for slower prometheus clients (default is 3s)
+  response_timeout = "15s"
+
+  ## Optional TLS Config
+  tls_ca = "/var/run/secrets/kubernetes.io/serviceaccount/ca.crt"
+  #tls_cert = /path/to/certfile
+  # tls_key = /path/to/keyfile
+  ## Use TLS but skip chain & host verification
+  insecure_skip_verify = true
+  #tagexclude = ["AgentVersion","AKS_RESOURCE_ID","ACS_RESOURCE_NAME", "Region", "ClusterName", "ClusterType", "Computer", "ControllerType"]
+  #[inputs.prometheus.tagpass]
```

Comment:
since we are not specified dimensions, are we collecting all the dimensions? are we planning expose the set of available dimensions in the config map so that customer can choose interesting ones within basic/advanced metrics if the dimensions are too many?

Reply:
Yes, we collect all dimensions, and workbook experience depend on them. Also, filtering out dimensions alone in a time series is dangerous. If your question is why are we not dropping time-series with certain dimension values, it depends on whats being collected v. whats being used. In this case everything thats exposed by them is for customer's consumption, so they are not filteredout/dropped.